typescript/api/services/VocabService.ts

```diff
@@ -146,6 +149,112 @@ export module Services {
       return response.data;
     }
 
+    public async findRecords(sourceType:string, brand:BrandingModel, searchString:string, start:number, rows:number): Promise<any> {
+
+      const report = sails.config.vocab[sourceType];
+
+      if (report.reportSource == 'database') {
+
+        let namedQueryConfig = await NamedQueryService.getNamedQueryConfig(brand, report.databaseQuery.queryName);
+
+        let configMongoQuery = namedQueryConfig.mongoQuery;
+        let collectionName = _.get(namedQueryConfig, 'collectionName', '');
+        let resultObjectMapping = _.get(namedQueryConfig, 'resultObjectMapping', {});
+        let brandIdFieldPath = _.get(namedQueryConfig, 'brandIdFieldPath', '');
+        let mongoQuery = _.clone(configMongoQuery);
+        let queryParams = namedQueryConfig.queryParams;
+        let paramMap = this.buildNamedQueryParamMap(report, searchString);
+
+        let dbResults = await NamedQueryService.performNamedQuery(brandIdFieldPath, resultObjectMapping, collectionName, mongoQuery, queryParams, paramMap, brand, start, rows);
+        if(report.resultObjectMapping) {
+          return this.getResultObjectMappings(dbResults,report);
+        } else {
+          return dbResults;
+        }
+      } else if (report.reportSource == 'solr') {
+        let solrQuery = this.buildSolrParams(brand, searchString, report, start, rows, 'json');
+        let solrResults = await this.getSearchService().searchAdvanced(report.searchQuery.searchCore, null, solrQuery);
+        if(report.resultObjectMapping) {
+          return this.getResultObjectMappings(solrResults,report);
+        } else {
+          return solrResults;
+        }
+      }
+    }
+
+    buildNamedQueryParamMap(report: any, searchString:string):any {
+      let paramMap = {}
+      if (report.queryField.type == 'text') {
+        paramMap[report.queryField.property] = searchString;
+      }
+      return paramMap;
+    }
+
+    private buildSolrParams(brand:BrandingModel, searchString:string, report:any, start:number, rows:number, format:string = 'json'):string {
+      let query = `${report.searchQuery.baseQuery}&sort=date_object_modified desc&version=2.2&start=${start}&rows=${rows}`;
+      query = query + `&fq=metaMetadata_brandId:${brand.id}&wt=${format}`;
+
+      if (report.queryField.type == 'text') {
+        let value = searchString;
+        if (!_.isEmpty(value)) {
+          let searchProperty = report.queryField.property;
+          query = query + '&fq=' + searchProperty + ':';
+          if(value.indexOf('*') != -1){
+            query = query + value.replaceAll('*','') + '*';
+          } else {
+            query = query + value + '*';
+          }
+        }
+      }
+
+      return query;
+    }
+
+    getResultObjectMappings(results: any, report: any) {
```

Comment:
Instead of report, use the query variable name and use the QueryConfig type instead of any